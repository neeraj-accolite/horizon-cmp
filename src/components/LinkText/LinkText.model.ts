export interface LinkTextProps {
  /**
   * The class applied to the parent element.
   */
  rootClass?: string;
  /**
   * Label for the Link Text
   */
  label: string;
  /**
   * Label Color
   */
  labelColor?: string;
  /**
   * Button size
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Font weight of the Link Text
   */
  fontWeight?: number;
  /**
   * To control underline under the text
   */
  underline?: boolean;
  /**
   * OnClick event of the Link Text
   */
  onClick?: () => void;
}
