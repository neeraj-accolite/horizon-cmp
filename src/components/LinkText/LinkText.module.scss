.text {
  @apply font-bold;
  @apply cursor-pointer;
}
