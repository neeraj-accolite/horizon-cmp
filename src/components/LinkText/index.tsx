export type { LinkTextProps } from "./LinkText.model";

export { default as LinkText } from "./LinkText";
