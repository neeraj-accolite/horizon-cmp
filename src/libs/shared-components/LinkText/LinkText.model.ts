export interface LinkTextProps {
  /**
   * The class applied to the parent element.
   */
  className?: string;
  /**
   * Define Label for the link text
   */
  label: string;

  /**
   * The URL the link points to.
   */
  href?: string;

  /**
   * Specifies the underline style for the link.
   */
  underline?: 'always' | 'hover' | 'none';

  /**
   * Click event handler for the link.
   */
  onClick?: () => void;
}
