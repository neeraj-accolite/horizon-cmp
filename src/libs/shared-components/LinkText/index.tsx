export { default as LinkText } from './LinkText';

export type { LinkTextProps } from './LinkText.model';
