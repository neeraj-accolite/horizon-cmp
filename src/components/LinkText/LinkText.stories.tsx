import LinkText from './LinkText';
import { Meta, StoryObj } from '@storybook/react';

export default {
  title: 'Components/LinkText',
  component: LinkText,
  parameters: { layout: 'centered' },
} as Meta<typeof LinkText>;

type Story = StoryObj<typeof LinkText>;

export const LinkedTextUnderlined: Story = {
  args: {
    label: 'Linked Text With Underline',
    onClick: () => console.log('Linked Text is clicked'),
  },
};

export const LinkedText: Story = {
  args: {
    label: 'Linked Text',
    onClick: () => console.log('Linked Text is clicked'),
    underline: false,
  },
};
