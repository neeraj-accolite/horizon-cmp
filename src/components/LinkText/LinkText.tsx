import React from 'react';
import { LinkTextProps } from './LinkText.model';
import { Link } from '@nextui-org/link';
import styles from './LinkText.module.scss';

const LinkText = ({
  label,
  onClick,
  rootClass,
  labelColor = 'inherit',
  size = 'medium',
  fontWeight = 700,
  underline = true,
  ...rest
}: LinkTextProps) => {
  const textStyle = {
    color: labelColor,
    fontWeight,
    fontSize: size === 'small' ? '12px' : size === 'medium' ? '16px' : '20px',
  };

  return (
    <Link
      {...rest}
      onClick={onClick}
      style={textStyle}
      className={`${styles.text} ${rootClass || ''}`}
      underline={underline ? 'always' : 'none'}
      aria-label={label}
    >
      {label}
    </Link>
  );
};

export default LinkText;
