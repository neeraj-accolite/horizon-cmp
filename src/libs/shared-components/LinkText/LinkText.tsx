'use client';

import React from 'react';
import Link from '../Link/Link';
import { LinkTextProps } from './LinkText.model';

const LinkText: React.FC<LinkTextProps> = ({
  label,
  className = '',
  underline = 'always',
  ...rest
}) => {
  return (
    <Link className={`${className}`} underline={underline} {...rest}>
      {label}
    </Link>
  );
};

export default LinkText;
