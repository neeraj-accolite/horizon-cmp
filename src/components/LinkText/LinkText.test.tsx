import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import LinkText from './LinkText';

describe('UI Component - LinkText', () => {
  test('renders the LinkText component with default props', async () => {
    const container = render(<LinkText label="Default LinkText" />);
    const linkText = await container.findByText('Default LinkText');
    expect(linkText).toBeInTheDocument();
    expect(linkText).toHaveStyle({
      color: 'inherit',
      fontSize: '16px',
      fontWeight: '700',
    });
  });

  test('onClick method is called when the LinkText is clicked', async () => {
    const handleClick = jest.fn();
    const container = render(
      <LinkText label="Clickable Text" onClick={handleClick} />,
    );
    const linkText = await container.findByText('Clickable Text');
    fireEvent.click(linkText);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('applies custom labelColor correctly', async () => {
    const container = render(
      <LinkText label="Colored Text" labelColor="#1976d2" />,
    );
    const linkText = await container.findByText('Colored Text');
    expect(linkText).toHaveStyle({ color: '#1976d2' });
  });

  test('renders with custom font size and weight', async () => {
    const container = render(
      <LinkText label="Large Bold Text" size="large" fontWeight={900} />,
    );
    const linkText = await container.findByText('Large Bold Text');
    expect(linkText).toHaveStyle({ fontSize: '20px', fontWeight: '900' });
  });

  test('applies rootClass when provided', async () => {
    const container = render(
      <LinkText label="Custom Class" rootClass="custom-class" />,
    );
    const linkText = await container.findByText('Custom Class');
    expect(linkText).toHaveClass('custom-class');
  });

  test('renders unchanged with default parameters', async () => {
    const container = render(
      <LinkText
        label="Snapshot Test"
        size="medium"
        fontWeight={700}
        underline={true}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders small text when size is small', async () => {
    const container = render(<LinkText label="Small Text" size="small" />);
    const linkText = await container.findByText('Small Text');
    expect(linkText).toHaveStyle({ fontSize: '12px' });
  });

  test('renders large text when size is large', async () => {
    const container = render(<LinkText label="Large Text" size="large" />);
    const linkText = await container.findByText('Large Text');
    expect(linkText).toHaveStyle({ fontSize: '20px' });
  });

  test('does not trigger onClick when not provided', async () => {
    const container = render(<LinkText label="No Click" />);
    const linkText = await container.findByText('No Click');
    fireEvent.click(linkText);
    expect(linkText).toBeInTheDocument();
  });
});
