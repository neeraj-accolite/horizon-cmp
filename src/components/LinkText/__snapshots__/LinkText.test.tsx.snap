// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component - LinkText renders unchanged with default parameters 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary underline hover:opacity-80 active:opacity-disabled transition-opacity underline-offset-4 text"
        role="link"
        style="font-weight: 700; font-size: 16px;"
        tabindex="0"
      >
        Snapshot Test
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary underline hover:opacity-80 active:opacity-disabled transition-opacity underline-offset-4 text"
      role="link"
      style="font-weight: 700; font-size: 16px;"
      tabindex="0"
    >
      Snapshot Test
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
